apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -1457,6 +1460,121 @@ def ingest_flat(
             parts_array.close()
             ids_array.close()
 
+    def ingest_vamana(
+        ctx,
+        index_group_uri: str,
+        source_uri: str,
+        source_type: str,
+        updates_uri: str,
+        vector_type: np.dtype,
+        external_ids_uri: str,
+        external_ids_type: str,
+        dimensions: int,
+        size: int,
+        batch: int,
+        config: Optional[Mapping[str, Any]] = None,
+        verbose: bool = False,
+        trace_id: Optional[str] = None,
+    ):
+        import numpy as np
+        import tiledb.cloud
+
+        logger = setup(config, verbose)
+        with tiledb.scope_ctx(ctx_or_config=config):
+            updated_ids = read_updated_ids(
+                updates_uri=updates_uri,
+                config=config,
+                verbose=verbose,
+                trace_id=trace_id,
+            )
+            group = tiledb.Group(index_group_uri)
+            parts_array_uri = group[PARTS_ARRAY_NAME].uri
```

Comment:
I think we should use the temp input arrays for this `INPUT_VECTORS_ARRAY_NAME, EXTERNAL_IDS_ARRAY_NAME`

Reply:
Thanks, good suggestion. Talked offline and then updated to use the same `"PARTIAL_WRITE_ARRAY_DIR": "temp_data",` which ivf_flat uses to write temp data because using `INPUT_VECTORS_ARRAY_NAME, EXTERNAL_IDS_ARRAY_NAME` is tricky with how we already create and write to it if `input_vectors` or `external_ids` are passed.